internal/controller/policyrecommendation_registrar.go

```diff
@@ -27,55 +29,71 @@ type PolicyRecommendationRegistrar struct {
-			// Object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		log.Error(err, "Error reading the object - requeue the request")
+	if !errors.IsNotFound(err) {
+		// Error occurred
+		logger.Error(err, "Failed to get Rollout. Requeue the request")
+		return reconcile.Result{}, err
+	}
+
+	// Check if Deployment exists
+	deployment := appsv1.Deployment{}
+	err = r.Client.Get(ctx, request.NamespacedName, &deployment)
+	if err == nil {
+		// Deployment exists, create policy recommendation
+		return r.createPolicyRecommendation(ctx, &deployment, logger)
+	}
+
+	if !errors.IsNotFound(err) {
+		logger.Error(err, "Failed to get Deployment. Requeue the request")
 		return reconcile.Result{}, err
 	}
 
+	logger.Info("Rollout or Deployment not found. It could have been deleted.")
+	return reconcile.Result{}, nil
+}
+
+func (r *PolicyRecommendationRegistrar) createPolicyRecommendation(ctx context.Context,
+	instance client.Object,
+	log logr.Logger) (ctrl.Result, error) {
 	// Check if a PolicyRecommendation object already exists
 	policy := &ottoscaleriov1alpha1.PolicyRecommendation{}
-	err = r.Client.Get(ctx, types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, policy)
+	err := r.Client.Get(ctx, types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}, policy)
 	if err == nil {
 		log.Info("PolicyRecommendation object already exists")
-		// PolicyRecommendation already exists, so we don't need to create one
 		return ctrl.Result{}, nil
 	} else if !errors.IsNotFound(err) {
-		// Error reading the object - requeue the request.
 		log.Error(err, "Error reading the object - requeue the request")
 		return ctrl.Result{}, err
 	}
 
 	log.Info("Creating a new PolicyRecommendation object")
-
-	// Create a new PolicyRecommendation object
+	log.Info(instance.GetObjectKind().GroupVersionKind().String())
 	newPolicyRecommendation := &ottoscaleriov1alpha1.PolicyRecommendation{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      instance.Name,
-			Namespace: instance.Namespace,
+			Name:      instance.GetName(),
+			Namespace: instance.GetNamespace(),
 			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(&instance, instance.GroupVersionKind()),
+				*metav1.NewControllerRef(instance, instance.GetObjectKind().GroupVersionKind()),
 			},
 		},
 		Spec: ottoscaleriov1alpha1.PolicyRecommendationSpec{
-			//TODO Set the fields in the policy spec as needed
-
+			WorkloadSpec: ottoscaleriov1alpha1.WorkloadSpec{Name: instance.GetName()},
+			//TODO Set the policy in the spec to safest policy
+			Policy:             ottoscaleriov1alpha1.Policy{},
+			QueuedForExecution: true,
+			GeneratedAt:        metav1.NewTime(time.Now()),
```

Comment:
This should be QueuedForExecutionAt since GeneratedAt is meant to record the time at which the recommendation is updated which serves a different purpose viz. aging trigger can look at GeneratedAt to identify stale recommendations.